server/store/views.py

```diff
@@ -47,6 +47,23 @@ class ProductPurchaseException(APIException):
     }
 
 
+def log_error(error_name, error_message, product):
+    # Log error by writing structured JSON. Can be then used with log-based alerting, metrics, etc.
+    print(
+        json.dumps(
```

Comment:
discussion: We're not adding the trace to this log, which means it won't have cross-reference to Cloud Tracing. I think that's fair. I think my preference would be to gain that by upgrading to a cloud logging library that has low overhead but handles that for us.

Reply:
I may be able manually add tracing per https://cloud.google.com/run/docs/logging#writing_structured_logs, but I would need to confirm I have access to the request in the view/serialiser contexts. I presumed the original ask in 207 didn't include tracing, but I can adapt this PR to suit.

Reply:
As discussed, this is fine for now. Adding in tracing is a future feature